weld/subsystem/src/main/java/org/jboss/as/weld/deployment/BeanDeploymentArchiveImpl.java

```diff
@@ -239,11 +239,12 @@ public boolean isAccessible(BeanDeploymentArchive target) {
             return true;
         }
 
+        final String thatIdentifier = that.getModule().getName();
         // basic check whether the module is our dependency
         for (DependencySpec dependency : module.getDependencies()) {
             if (dependency instanceof ModuleDependencySpec) {
                 ModuleDependencySpec moduleDependency = (ModuleDependencySpec) dependency;
-                if (moduleDependency.getIdentifier().equals(that.getModule().getIdentifier())) {
+                if (moduleDependency.getName().equals(thatIdentifier)) {
```

Comment:
This is the only part where I am not certain - the comparison takes the name of `ModuleDependencySpec` and that of a `Module`. But I assume that if their identifiers were comparable and those were replaced by String names, then the names should be comparable as well. Feel free to prove me wrong.

Reply:
@manovotn LGTM.  I'll just assume the existing equals is valid. If not, that's not the concern of this PR. ;) 

Re the change in both classes getIdentifier() is deprecated with javadoc saying to use getName(). And internally both implement getIdentfier() as ModuleIdentifier.fromString(name). So this change is valid according to both the implementation and the javadoc.